addons/api/addon/models/host-catalog.js

```diff
@@ -62,22 +64,22 @@ export default class HostCatalogModel extends GeneratedHostCatalogModel {
    * @type {string}
    */
   get credentialType() {
-    if (!this.#credentialType) {
+    if (!this.credentialTypeValue) {
       if (DYNAMIC_CREDENTIAL_FIELDS.some((field) => this[field])) {
-        this.#credentialType = TYPE_CREDENTIAL_DYNAMIC;
+        return TYPE_CREDENTIAL_DYNAMIC;
```

Comment:
None blocking comment: 
The previous implementation checks if `this.credentialType` exists.
If it doesn't, it sets a value.
At the end, it always returns `this.credentialType`

The new implementation checks if `this.credentialType` exists.
If it doesnt', returns a value. BUT, and here is my questions, never sets the `credentialType` @tracked property. I have nothing against, I would like to learn Why we change the approach on this?
If it does, returns the `this.credentialType`

Reply:
Great question, this getter shouldn't set the value of `credentialType` once it was changed to be `tracked` because we get [this error](https://github.com/hashicorp/boundary-ui/pull/2711#discussion_r1987705022). It also doesn't need to set the value for `credentialType` because the getter takes care of returning the correct type on initial load (when the private variable `credentialType` is undefined).

Reply:
@lisbet-alvarez thanks for your detailed answer! makes sens 😃